server.js

```diff
@@ -19,5 +20,48 @@ const messages = [welcomeMessage];
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
+app.post("/messages", (req, res) => {
+  //validate our input
+  if (!req.body.from || !req.body.text) {
+    res.status(400).send("from must be valid");
+    return;
+  }
+
+  // build our new resourse
+  const newMessage = req.body;
```

Comment:
You shouldn't put the entire body into the newMessage object, as there may be other fields in the req.body object that you haven't validated